connectors/es/management_client.py

```diff
@@ -193,3 +198,49 @@ async def get_connector_secret(self, connector_secret_id):
             )
         )
         return secret.get("value")
+
+    async def create_connector_secret(self, secret_value):
+        secret = await self._retrier.execute_with_retry(
+            partial(
+                self.client.perform_request,
+                "POST",
+                "/_connector/_secret",
+                headers={
+                    "accept": "application/json",
+                    "content-type": "application/json",
+                },
+                body={"value": secret_value},
+            )
+        )
+        return secret.get("id")
+
+    async def generate_and_store_api_key(self, index_name):
+        """
+        Generates an API key for the chosen index and stores it in secrets storage.
+
+        This function should only be called for native connectors
+        """
+        role_descriptors = {
+            f"{alphanumericize_string(index_name)}-connector-role": {
+                "cluster": ["monitor"],
+                "index": [
+                    {
+                        "names": [
+                            index_name,
+                            f"{CONNECTORS_ACCESS_CONTROL_INDEX_PREFIX}{index_name}",
+                            f"{CONNECTORS_INDEX}*",
+                        ],
+                        "privileges": ["all"],
+                    }
+                ],
+            }
+        }
+        api_key_result = await self._retrier.execute_with_retry(
+            partial(
+                self.client.security.create_api_key,
+                name=f"{index_name}-connector",
```

Comment:
Why `-connector`?

Reply:
The API key name is arbitrary and is only useful for seeing the API key in the Kibana UI. This is just identical implementation to Kibana.
I think the `-connector` suffix is because the API key will have access to the connector's index, ACL index, and the `.elastic-connectors` index. So it's a very connector-centric API key.